pkgs/development/libraries/mkldnn/default.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchFromGitHub, cmake, substituteAll, bash, autoPatchelfHook }:
+
+stdenv.mkDerivation rec {
+  pname = "mkldnn";
+  version = "1.0.2";
+
+  src = fetchFromGitHub {
+    owner = "intel";
+    repo = "mkl-dnn";
+    rev = "v${version}";
+    sha256 = "0lf14a4j511pydyz78wwbl6hifbq86m3dy119iss9cvz8zzrdyvw";
+  };
+  outputs = [ "out" "dev" "doc" ];
+  patches = [ (substituteAll {
+    src = ./bash-to-sh.patch;
+    inherit bash;
+  }) ];
+
+  nativeBuildInputs = [ cmake ];
```

Comment:
`mkl-dnn` can optionally be built against the `mkl` package:
https://github.com/intel/mkl-dnn/blob/master/cmake/MKL.cmake

This has a proprietary license so it's not built/distributed by default in NixPkg and Hydra, but it is packaged.  For users who want to use it it might be nice to give them an optional switch, like we do with `numpy`:
https://nixos.org/nixpkgs/manual/#how-to-use-intels-mkl-with-numpy-and-scipy

Reply:
I'm slightly confused, should this have blas as an input?